src/pages/homepage/Homepage.jsx

```diff
@@ -1,44 +1,30 @@
+import { useState, useEffect } from 'react'
+import axios from 'axios'
 import '../../styles/index.css'
 import { image, center as heroImage } from '../../assets/images/index'
 import { Navbar } from '../../components/navbar/Navbar.jsx'
-import { Link } from 'react-router-dom'
-
+import { HomeProductCard } from './HomeProductCard'
 export function Homepage() {
+  const [categories, setCategories] = useState([])
+  useEffect(() => {
+    try {
+      ;(async () => {
+        const { data } = await axios.get('/api/categories')
+        setCategories(data.categories)
+      })()
+    } catch (e) {
+      console.log(e)
+    }
```

Comment:
you should show the error in the ui so the user also know whats happening.

Reply:
Thanks for suggesting this, I have now made a error state and displaying it on the UI